File: warmups/03.robot-to-origin/main.cpp
class Solution {
public:
    bool judgeCircle(string moves) {
        int x = 0;
        int y = 0;

        for (char move : moves) {
            switch (move) {
                case 'L': x--; break;
                case 'R': x++; break;
                case 'U': y++; break;
                case 'D': y--; break;
            }
        }

        return x == 0 && y == 0;
    }
};

File: warmups/03.robot-to-origin/main.py
class Solution:
    def judgeCircle(self, moves: str) -> bool:
        x, y = 0, 0

        for move in moves:
            if move == 'L':
                x -= 1
            elif move == 'R':
                x += 1
            elif move == 'U':
                y += 1
            else:
                y -= 1

        return x == 0 and y == 0
